### PR TITLE
Retry google bucket creation (resolves #885)

### LINKED ISSUE #885: Too many requests error from Google job store
See Jenkins build [here](http://jenkins.cgcloud.info/job/toil-pull-requests/1326/).

### PR DESCRIPTION
**Pull request by cket**

### CODE REVIEW

File: src/toil/jobStores/googleJobStore.py [hunk 1]
```diff
@@ -18,6 +18,18 @@
 
 GOOGLE_STORAGE = 'gs'
 
+def _retryCreateBucket(uri, headers):
```
Comment (hannes-ucsc) [inline src/toil/jobStores/googleJobStore.py]: This should be a static or class method of the GoogleJobStore class.

File: src/toil/jobStores/googleJobStore.py [hunk 2]
```diff
@@ -18,6 +18,18 @@
...
+    bucket = None
+    while not bucket:
+        try:
+            bucket = uri.create_bucket(headers=headers)
+        except boto.exception.GSResponseError as e:
+            if e.status == 429:
+                time.sleep(1)
```
Comment (hannes-ucsc) [inline src/toil/jobStores/googleJobStore.py]: Change to 10s.

File: src/toil/jobStores/googleJobStore.py (hunk 1 above)
Comment (cket) [inline src/toil/jobStores/googleJobStore.py]: My reasoning was that there is nothing in the method tied to the job store so it logically doesn't belong inside the object. The import/export tests also use the method to create a non-job store bucket.

Comment (hannes-ucsc) [inline src/toil/jobStores/googleJobStore.py]: Still, please make it a static method.

**Comment by hannes-ucsc** [comment]
Tests fail due to stupid refactoring error. Please fix AS SOON AS POSSIBLE. Pretty much everyone is blocked on this.

**Comment by cket** [comment]
Build failed on unrelated test failure.

**Comment by cket** [comment]
Jenkins, test this please.

**Comment by cket** [comment]
Build failed on:
src.toil.test.sort.sortTest.SortTest.testAzureSingle
src.toil.test.sort.sortTest.SortTest.testGoogleMesos
src.toil.test.src.jobCacheTest.AzureJobStoreCacheTest.testDeleteLocalFile
src.toil.test.src.jobCacheTest.AzureJobStoreCacheTest.testReadCacheMissFileFromJobStoreWithCachingReadFile

Both sort tests failed on stats and logging aggregator process exiting with nonzero status codes. Failures don't appear to be related to this PR.

**Comment by cket** [comment]
Jenkins, test this please.

**Comment by hannes-ucsc** [comment]
I've merged this despite the single test failure in the most recent build. That failure is already tracked as #760.

**Pull request merged**